Hardening proc macro path resolution broke cargo overrides

This broke all my usages of bevy derive with messages like this:

```
error[E0433]: failed to resolve: use of undeclared crate or module `bevy_reflect`
   --> C:\Users\pcwal\Source\Rust\leafwing-input-manager\src\lib.rs:144:41
    |
144 | #[derive(Debug, Clone, Copy, PartialEq, Reflect, Serialize, Deserialize)]
    |                                         ^^^^^^^ use of undeclared crate or module `bevy_reflect`
    |
    = note: struct `crate::user_input::trait_reflection::tripleaxislike::TypeRegistration` exists but is inaccessible
    = note: this error originates in the derive macro `Reflect` (in Nightly builds, run with -Z macro-backtrace for more info)
```

and

```
error[E0277]: `user_input::gamepad::GamepadStick` can not provide dynamic type information through reflection
   --> C:\Users\pcwal\Source\Rust\leafwing-input-manager\src\user_input\gamepad.rs:548:23
    |
548 | impl DualAxislike for GamepadStick {
    |                       ^^^^^^^^^^^^ the trait `bevy::bevy_reflect::Typed` is not implemented for `user_input::gamepad::GamepadStick`
```

For reference, I'm using Cargo overrides to redirect Bevy 0.16.0 to the local repo.

_Originally posted by @pcwalton in https://github.com/bevyengine/bevy/issues/17330#issuecomment-2646851696_

--- a/crates/bevy_macro_utils/Cargo.toml
+++ b/crates/bevy_macro_utils/Cargo.toml
@@ -12,7 +12,7 @@ keywords = ["bevy"]
 syn = "2.0"
 quote = "1.0"
 proc-macro2 = "1.0"
-cargo-manifest-proc-macros = "0.3.3"
+cargo-manifest-proc-macros = "0.3.4"
 
 [lints]
 workspace = true

--- a/crates/bevy_macro_utils/src/bevy_manifest.rs
+++ b/crates/bevy_macro_utils/src/bevy_manifest.rs
@@ -48,8 +48,7 @@ impl BevyManifest {
     /// Returns the path for the crate with the given name.
     pub fn get_path(&self, name: &str) -> syn::Path {
         self.maybe_get_path(name)
-            //.expect("Failed to get path for crate")
-            .unwrap_or_else(|_err| Self::parse_str(name))
+            .expect("Failed to get path for crate")
     }
 
     /// Attempt to parse the provided [path](str) as a [syntax tree node](syn::parse::Parse)

--- a/tests-integration/simple-ecs-test/Cargo.toml
+++ b/tests-integration/simple-ecs-test/Cargo.toml
@@ -2,5 +2,10 @@
 name = "simple-ecs-test"
 edition = "2021"
 
+# We depend on bevy in both normal and dev dependencies to verify that the proc macros still work.
+
 [dependencies]
+bevy = { path = "../../" }
+
+[dev-dependencies]
 bevy = { path = "../../" }

